src/Microsoft.Cci.Extensions/Extensions/CSharp/CSharpCciExtensions.cs

```diff
@@ -333,6 +333,30 @@ public static bool IsDestructor(this IMethodDefinition methodDefinition)
             return false;
         }
 
+        public static bool IsDispose(this IMethodDefinition methodDefinition)
+        {
+            if (methodDefinition.Name.Value != "Dispose")
+            {
+                return false;
+            }
+
+            if (methodDefinition.ParameterCount == 0 && methodDefinition.Visibility == TypeMemberVisibility.Public &&
```

Comment:
>f (methodDefinit [](start = 13, length = 16)

Is there anyway (in CCI) to detect if the method is part of IDisposable interface implementation?

Reply:
@weshaggard and I talked about this and we said it was not worth the effort since we would need to traverse the whole tree as it could come from a base class, not necessarily direct implementation.

Reply:
I think there is a way through the Type object:

https://stackoverflow.com/questions/503263/how-to-determine-if-a-type-implements-a-specific-generic-interface-type

what do you guys think?

Reply:
I think this should not be much to implement. is it possible someone will have Dispose and not implementing IDisposable?

Reply:
You can have it, but for our usages I don't think we would have it like that. I think for our convention, having a Dispose method should mean being IDisposable

Reply:
Yes it is possible but for our scenarios I think it is OK to just ensure the method is called Dispose and returns void.